### ISSUE TITLE
[Suggestion] Be able to either swap or remove same items in hand without needing an emtpy hand

### ISSUE THREAD

**Issue by DWX12** [codetaylor/pyrotech#112]
This is a suggestion for a quality of life addition to the mod. Whenever I'm using a drying rack or, especially, the worktable, I usually end up having a full hotbar and find myself swapping between ingredients and tools. However, whenever I miss-click and add an item that I was intending to, I have to clear out a space in my hotbar in order to retrieve the incorrect item.

I would like to be able to grab items of the same type while holding it in addition to being able to swap items from the hand to an slot on a machine, such as the worktable, with an existing item on the machine. This would just simplify crafting and working with machines in a much more pleasing manner.

This is a great mod and the curseforge page is superbly done. Thank you for all that you put into this mod.

**Comment by codetaylor** [comment]
You can sneak + scroll to insert / remove items even if your hand is full.

**Comment by Foreck1** [comment]
I must agree with the original suggestion here. From a basic gameplay approach, the way it works right now hinders quick management of blocks and the flow of the game quite a lot. Ideally, it would go like this:

- Right-click with item on empty slot: puts 1 unit of the item on empty slot
- Right-click with item on slot holding the same item: adds 1 unit more of said item
- Right-click with item on slot holding different item: removes 1 unit of hold item
- Right-click with empty hand on slot holding item: removes 1 unit of hold item
- **Shift** + any of the actions above modify the behavior to operate on full stacks instead of 1 item at a time

**Comment by codetaylor** [comment]
Shift + right-click with a held block should not override the default Minecraft behavior. When I shift + right-click with a block, I expect the action will override any special tile or block interaction and place a block.
